### PR TITLE
fix: re-enable react strict mode

### PR DESCRIPTION
**Pull request by wittjosiah**
This also partially addresses #2627. Support for handling `forwardRef` will be addressed in a followup.

### CODE REVIEW

**Comment by wittjosiah** [comment]
Current dependencies on/for this PR:
* main
  * **PR #2621** <a href="https://app.graphite.dev/github/pr/dxos/dxos/2621" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dxos/dxos/2621?utm_source=stack-comment).

File: packages/experimental/kai/src/containers/deprecated/TaskList.tsx [hunk 1]
```diff
@@ -103,35 +103,33 @@ export const TaskItem: FC<{
...
-        slots={{
-          ...slots,
-          listItem: {
-            selectableCheckbox: {
-              className:
-                'radix-state-checked:bg-white radix-state-unchecked:bg-white radix-state-checked:border radix-state-unchecked:border border-primary-600 text-primary-600'
-            }
+}> = observer(({ task, onDelete, onSave, slots }) => {
+  // TODO(wittjosiah): Remove?
+  // useReactorContext({
```
Comment (wittjosiah) [inline packages/experimental/kai/src/containers/deprecated/TaskList.tsx:108]: @dmaretskyi after you removed this, this component didn't build. How should this behaviour be replicated?

File: packages/sdk/react-client/src/client/ClientContext.test.tsx [hunk 2]
```diff
@@ -82,23 +88,19 @@ describe('ClientProvider', () => {
...
-  });
-
   test('Renders with children', async () => {
     render(
       <ClientProvider client={client}>
         <TestComponent />
       </ClientProvider>
     );
 
-    // TODO(wittjosiah): This no longer works. Because the client status becomes active during initialization?
```
Comment (wittjosiah) [inline packages/sdk/react-client/src/client/ClientContext.test.tsx:101]: The `afterEach` was breaking this. The ClientProvider destroys client on unmount so it's unneeded.

File: packages/sdk/react-client/src/testing/ClientSpaceDecorator.tsx [hunk 3]
```diff
@@ -5,18 +5,63 @@
...
 import { log } from '@dxos/log';
-import { useAsyncEffect } from '@dxos/react-async';
 import { Loading } from '@dxos/react-components';
 
 import { ClientProvider } from '../client';
 
+const testBuilder = new TestBuilder();
+const services = () => testBuilder.createClientServicesHost();
+
+// TODO(wittjosiah): Generates warning `No peers to notarize with` during invitation, but retry succeeds.
```
Comment (wittjosiah) [inline packages/sdk/react-client/src/testing/ClientSpaceDecorator.tsx:22]: @dmaretskyi @richburdon if you have any thoughts on why this is let me know

**Review by dmaretskyi — APPROVED** [review]
useSubscription needs fixing, but I can do that separately. LGTM otherwise

File: packages/experimental/kai/src/containers/deprecated/TaskList.tsx (hunk 1 above)
Comment (dmaretskyi) [reply packages/experimental/kai/src/containers/deprecated/TaskList.tsx:108]: Use subscription would work here. But perhaps there’s another way to rewrite this code to not need the hook?

File: packages/sdk/react-client/src/testing/ClientSpaceDecorator.tsx (hunk 3 above)
Comment (dmaretskyi) [reply packages/sdk/react-client/src/testing/ClientSpaceDecorator.tsx:22]: Yeah, warning is a non issue, we could hide it

**Pull request merged**